patch-test.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-env node, es6 */
+/* eslint-disable no-console */
+/* eslint-disable es5/no-destructuring */
+/* eslint-disable es5/no-template-literals */
+/* eslint-disable es5/no-es6-methods */
+/* eslint-disable es5/no-shorthand-properties */
+/* eslint-disable es5/no-block-scoping */
+
+/**
+ * Creates a file patch-test-loader.js with the necessary import statements
+ * to test the changes made in a pull request or in the current working state
+ *
+ * How to use:
+ * - Run `npm run patchtest` which generates patch-test-loader.js
+ * - Set up a localhost server (such as by using server.js or by running
+ *   php -S localhost:5500) and load patch-test-loader.js in the wiki environment,
+ *   using the browser console or the common.js page.
+ */
+
+const fs = require('fs');
+const {execSync} = require('child_process');
+
+const server = 'http://127.0.0.1:5500';
```

Comment:
Why 5500?  I always thought 8000 or 8080 were more standard for one-off local hostings?  Is this a V$code thing?

(I thought I asked this yesterday, but must've gotten gobbled up somewhere.)

Reply:
I don't know about 8000, but 8080 is the port used by Apache Tomcat servers and so it frequently ends up being used for HTTP applications. In short, they're used for serious stuff :)

Reply:
Well... under 1024 are the reserved, "serious" ones.  80 is http, which I believe is the reason why historically 8000 and 8080 are both used for various unofficial hostings and one-offs, and as an http alternative (8008 too).  None of it matters, really, unless the person is testing *and* serving something else, and even then that's on them!  There's thousands to choose from, so it's largely a moot consideration, but 8000/8080 have been used for hosting and casting for *decades*.  As an example, it's the default for python's `SimpleHTTPServer`.

Reply:
Well, using a standard port like 8000/8080 does not make sense as these are the ones that are most likely to be already in use. They are used for serious hostings (my college's wifi login page used to on 8080, until they switched to sophos). This is I believe because usually the decision isn't really in their hand -- it's determined by the server software or the service provider. Eg. on Toolforge, the nginx proxy decides what port to use and puts that out as an environment variable [which the tool devs use](https://github.com/siddharthvp/toolforge-node-app-base/blob/master/server.js#L21). 

Arguing over what port to use for a service that would be used only by the person hosting it is one of the most pointless things we can do. Since we can choose anything, it probably makes sense to use a completely random port, eg. [xfdcloser uses](https://github.com/evad37/xfdcloser/blob/master/server.js) 8125 and [redwarn uses](https://gitlab.com/redwarn/redwarn-web/-/blob/master/CONTRIBUTING.md) 9696.

Reply:
Hey man sorry if it came off like I was trying to argue or bikeshed, I really wasn't trying to.  We're talking past each other a bit here, but I really was just curious where 5500 came form.  It's a nice round number and I'm used to seeing 8000for personal localhostings, so I was wondering what led you to use it.  You vaguely referenced VScode, so I thought maybe it was connected to that?  I've [no issue](https://github.com/azatoth/twinkle/pull/944#pullrequestreview-446904635) with this as is, and sorry for making you feel like I did.  Thanks for putting this together, it's neat!